src/test_workflow/perf_test/perf_single_node_cluster.py

```diff
@@ -0,0 +1,44 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import os
+
+from test_workflow.perf_test.perf_test_cluster import PerfTestCluster
+
+
+class PerfSingleNodeCluster(PerfTestCluster):
+    """
+    Represents a performance single node test cluster. This class deploys the opensearch bundle with CDK.
+    """
+    def __init__(self, bundle_manifest, config, stack_name, cluster_config, current_workspace):
+        assert cluster_config.is_single_node_cluster, "Cluster is not a single node cluster"
+        work_dir = os.path.join(current_workspace, "opensearch-cluster", "cdk", "single-node")
+        super().__init__(bundle_manifest, config, stack_name, cluster_config, current_workspace, work_dir)
+
+    def create_endpoint(self, cdk_output):
+        scheme = "https://" if self.cluster_config.security else "http://"
+        private_ip = cdk_output[self.stack_name]["PrivateIp"]
+        public_ip = cdk_output[self.stack_name].get("PublicIp", None)
+        self.is_endpoint_public = public_ip is not None
+        host = public_ip if public_ip is not None else private_ip
+        assert host is not None, "Unable to fetch the cluster endpoint from cdk output"
```

Comment:
I think this error should happen at runtime, there's a legit cdk output that might be missing this data, so I think it's not an assert, but a normal exception. And would need a test that it's raised.

Reply:
Sure. Changed to RuntimeError and added the tests for it.